### DISCUSSION TITLE
Unique field validation filter for currently authenticated user

### DISCUSSION THREAD

**Discussion by devjabeer** [directus/directus#5950] [Archive]
Hi,
I want to create a field validation, inside custom permission for 'create' for a collection, that field should be unique to currently authenticated user, but it can be duplicate for different users. 
Example: I have a collection name "Pages" with a field name "name", same user cannot create items in pages with duplicate name, say a user cannot create page with name "my page" twice but two different user can have items with name "my page".

How to approach this case, I was thinking to put custom field validation in place but don't know to form this logic using filter rules. Or another option is to set this logic in front end side of the application.

**Comment by philleepflorence** [comment]
It sounds like you want to have the fields (`name`, `user_created`) to be a unique index in the `pages` collection.

That translates to the following, if using [MySQL or MariaDB](https://www.w3schools.com/sql/sql_unique.asp), and you have direct access to the [Database](https://docs.directus.io/concepts/databases/):
```
ALTER TABLE  `pages` ADD CONSTRAINT  `pages_user_name` UNIQUE KEY (`name`, `user_created`)
```

To make only a single field unique by using the Directus Settings: [Directus Docs: Fields - Schema Setup](https://docs.directus.io/guides/fields/#schema-setup)

**Reply by devjabeer** [reply]
I am using Postgresql and docker directus. How to setup custom schema in this case? any idea?

**Reply by philleepflorence** [reply]
I am not familiar with Docker, but to add a unique index or constraint for [PostgreSQL](https://www.postgresqltutorial.com/postgresql-indexes/postgresql-unique-index/):
```
CREATE UNIQUE INDEX  pages_user_name ON  pages(name, user_created)
```
I am sure there are community examples or tips for Docker, PostgreSQL Schema, and Unique Columns.

**Comment by devjabeer** [comment]
We require more dynamic values as discussed here
https://github.com/directus/directus/discussions/2993
